backend/src/main/kotlin/org/loculus/backend/service/submission/CompressionService.kt

```diff
@@ -23,15 +23,23 @@ enum class CompressionAlgorithm(val extension: String) {
 }
 
 @Service
-class CompressionService(private val backendConfig: BackendConfig) {
+class CompressionService(
+    private val backendConfig: BackendConfig,
+    private val compressionDictService: CompressionDictService,
+) {
+
+    fun compressOriginalSequence(sequenceData: GeneticSequence, organism: Organism) = compress(
+        sequenceData,
+        compressionDictService.getDictForUnalignedSequence(organism),
+    )
 
-    fun compressNucleotideSequence(
+    private fun compressNucleotideSequence(
         uncompressedSequence: GeneticSequence,
         segmentName: String,
         organism: Organism,
     ): CompressedSequence = compress(
         uncompressedSequence,
-        getDictionaryForNucleotideSequenceSegment(segmentName, organism),
+        compressionDictService.getDictForSegmentOrGene(organism, segmentName),
```

Comment:
(This likely precedes this PR) I find it confusing that functions sometimes use `organism` first, sometimes `segmentName`. I think `organism` then `segmentName` makes more sense - we might want to regularize this across the codebase (maybe in different refactoring PR).